### PR TITLE
Use poetry for package management.

### PR DESCRIPTION
**Pull request by valberg**
This should be merged before #178 

A lot of credit goes to @MrCordeiro and his changes in https://github.com/MrCordeiro/django-invitations

### CODE REVIEW

**Review by MrCordeiro — APPROVED** [review]

**Comment by jezdez** [comment]
@valberg @MrCordeiro FWIW I haven't tested the Jazzband index with Poetry, so this may or may not fail.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -23,16 +23,14 @@ jobs:
...
       - name: Upload packages to Jazzband
         if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
-        uses: pypa/gh-action-pypi-publish@master
-        with:
-          user: jazzband
-          password: ${{ secrets.JAZZBAND_RELEASE_KEY }}
-          repository_url: https://jazzband.co/projects/django-invitations/upload
+        run: |
+          poetry config repositories.jazzband https://jazzband.co/projects/django-invitations/upload
+          poetry config http-basic.jazzband jazzband ${{ secrets.JAZZBAND_RELEASE_KEY }}
```
Comment (jezdez) [inline .github/workflows/release.yml:36]: In fact, this won't actually upload the file, @valberg! It just configures it.

Comment (MrCordeiro) [reply .github/workflows/release.yml:36]: @valberg, we're missing a publish step:

```yml
    - name: Upload packages to Jazzband
        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
        uses: pypa/gh-action-pypi-publish@master
        with:
          user: jazzband
          password: ${{ secrets.JAZZBAND_RELEASE_KEY }}
          repository_url: https://jazzband.co/projects/wagtailmenus/upload 
```

(I'm curious to see if building it with poetry will work. Perhaps using `poetry publish` instead could work too?)

Comment (jezdez) [reply .github/workflows/release.yml:36]: https://github.com/jazzband/django-invitations/pull/195 fixed this FWIW

**Pull request merged**